UniAtHome/UniAtHome.WebAPI/Controllers/UsersController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using UniAtHome.BLL.Interfaces;
+using UniAtHome.DAL.Entities;
+using UniAtHome.WebAPI.Models.UsersRequests;
+
+namespace UniAtHome.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly UserManager<User> userManager;
+
+        private readonly IAuthTokenGenerator tokenGenerator;
+
+        public UsersController(UserManager<User> userManager, IAuthTokenGenerator tokenGenerator)
+        {
+            this.userManager = userManager;
+            this.tokenGenerator = tokenGenerator;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<object> Register([FromBody] RegistrationRequest registerModel)
+        {
+            var user = new User
```

Comment:
Please, don't create User instance here. We shouldn't use Entity classes in Controllers. Create DTO object i you need it.
In future wi will use AutoMapper

Reply:
UserManager works specifically with User objects as well the UserRepository does. Shall I remove all the code and put it inside BLL?

Reply:
All business logic should be in bll. Managing users refers to business logic.